falcon/asgi/ws.py

```diff
@@ -728,16 +754,12 @@ async def _pump(self):
                 self._pop_message_waiter = None
 
 
-def check_support_reason(asgi_ver):
+@misc._lru_cache_for_simple_logic(maxsize=16)
+def _supports_reason(asgi_ver: str):
```

Comment:
any reason for making this private?

Reply:
It was not documented publicly. We have an issue in the 4.0 milestone to privatize as much as possible. Do we want this function in Falcon's public interface with SemVer guarantees? I don't think so.

Reply:
Not sure how to best achieve that. Maybe indicating the public modules/packages is better? 

but we cab decide that at some other time